### PR TITLE
MDP-195: buttons and input component

### PR DESCRIPTION
**Pull request by drerchi**

### CODE REVIEW

**Review by garySiladi — DISMISSED** [review]

**Review by garySiladi — COMMENTED** [review]
x

File: src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { FC } from "react";
+
+interface IProps {
+  inverted?: boolean;
+  text?: string;
+  color?: boolean;
+  size?: boolean;
+}
```
Comment (pggl) [inline src/components/Button.tsx:6]: I think text props is required and other props need default value

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      color ? "white" : "black"
+    } font-lato text-base hover:shadow-lg hover:bg-dark-black active:opacity-50 mr-20 font-bold bg-${
+      inverted ? "black" : "white"
+    }`}
+    style={{
+      borderRadius: "12px",
+      borderColor: "black",
+      borderWidth: "1px",
+      outline: "0"
+    }}
```
Comment (pggl) [inline src/components/Button.tsx]: why is here this style, use tailwind

File: src/components/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+
+const StyledInput = styled.div`
+  width: 391px;
```
Comment (pggl) [inline src/components/Input.tsx]: width should be 100%

File: src/components/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  height: 40px;
+`;
+
+export const Input: FC = () => (
+  <StyledInput>
+    <p className="text-sm text-grey font-lato mb-2 mt-3">Email</p>
```
Comment (pggl) [inline src/components/Input.tsx]: why is "Email" hardcoded

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx (hunk 1 above)
Comment (drerchi) [reply src/components/Button.tsx:6]: done

File: src/components/Button.tsx (hunk 2 above)
Comment (erikruszinka) [reply src/components/Button.tsx]: Not working with tailwind

File: src/components/Input.tsx (hunk 3 above)
Comment (drerchi) [reply src/components/Input.tsx]: done

File: src/components/Input.tsx (hunk 4 above)
Comment (drerchi) [reply src/components/Input.tsx]: done with props

File: src/components/Button.tsx (hunk 2 above)
Comment (drerchi) [reply src/components/Button.tsx]: done

File: tailwind.js [hunk 5]
```diff
@@ -54,7 +54,8 @@ let colors = {
   maize: "#ffc450",
   purple: "#db3eb1",
   red: "#f93822",
-  white: "#ffffff"
+  white: "#ffffff",
```
Comment (garySiladi) [inline tailwind.js]: this file shouldn't be re-prettifyied.
I opened a PR which excludes this file from prettier formatting in https://github.com/garySiladi/chat-project/pull/18/files which will fix this
at the moment try to reset this file, momentarily disable the autoformat prettier extension, add your changes and commit

File: src/components/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  name: string;
+}
+
+export const Input: FC<IProps> = ({ name }) => (
+  <div>
+    <p className="text-sm text-grey font-lato mb-2 mt-3">{name}</p>
+    <input
+      type="text"
+      placeholder="Zadajte Váš email"
+      className={`w-full h-10 font-lato text-base text-sm hover:shadow-lg pl-4`}
```
Comment (garySiladi) [inline src/components/Input.tsx]: no need for the brace string literal syntax, just simply ""

File: src/components/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { FC } from "react";
+
+interface IProps {
+  inverted?: boolean;
```
Comment (garySiladi) [inline src/components/Button.tsx]: this prop could be called `isInverted`, so we know from the sound of it, that we're dealing with a boolean.
It's a good naming convention.

File: src/components/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  text: string;
+  isColorWhite?: boolean;
+  size?: boolean;
```
Comment (garySiladi) [inline src/components/Button.tsx]: the name `size` doesn't describe precisely that we're dealing with the boolean.
try naming it something like `isBig` or similar.

File: src/components/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export const Button: FC<IProps> = ({
+  inverted = false,
+  text,
+  isColorWhite = false,
+  size = false
+}) => (
+  <button
+    className={`mt-1 h-10 w-${size ? "10" : "48"} text-${
```
Comment (garySiladi) [inline src/components/Button.tsx]: I would create the substring in a separate function so it becomes easier to read

**Review by garySiladi — CHANGES_REQUESTED** [review]
some changes required

File: src/components/Input.tsx (hunk 6 above)
Comment (drerchi) [reply src/components/Input.tsx]: done

File: src/components/Button.tsx (hunk 7 above)
Comment (drerchi) [reply src/components/Button.tsx]: done

File: src/components/Button.tsx (hunk 8 above)
Comment (drerchi) [reply src/components/Button.tsx]: done

**Review by garySiladi — DISMISSED** [review]

File: src/components/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  isInverted?: boolean;
+  text: string;
+  isColorWhite?: boolean;
+  isSmall?: boolean;
+}
+
+export const Button: FC<IProps> = ({
+  isInverted = false,
+  text,
+  isColorWhite = false,
```
Comment (pggl) [inline src/components/Button.tsx]: why this props? we have this information from isInverted

File: src/components/Button.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  isSmall = false
```
Comment (pggl) [inline src/components/Button.tsx]: we don't need this, this should be dynamic

File: src/components/Input.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from "react";
+
+interface IProps {
+  name: string;
+}
+
+export const Input: FC<IProps> = ({ name }) => (
+  <div>
+    <p className="text-sm text-grey font-lato mb-2 mt-3">{name}</p>
```
Comment (pggl) [inline src/components/Input.tsx]: use label tag for this

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx (hunk 10 above)
Comment (drerchi) [reply src/components/Button.tsx]: done

File: src/components/Button.tsx (hunk 11 above)
Comment (drerchi) [reply src/components/Button.tsx]: done

File: src/components/Input.tsx (hunk 12 above)
Comment (drerchi) [reply src/components/Input.tsx]: done

File: src/components/Input.tsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+interface IProps {
+  name: string;
+}
+
+export const Input: FC<IProps> = ({ name }) => (
+  <div>
+    <label className="text-sm text-grey font-lato mb-2 mt-3">{name}</label>
+    <input
+      type="text"
+      placeholder="Zadajte Váš email"
```
Comment (pggl) [inline src/components/Input.tsx]: we need props for this

File: src/stories/index.tsx [hunk 14]
```diff
@@ -7,8 +7,19 @@ import { CancelReservationPopup } from "../components/Popup/Storybook/CancelRese
 import { Header } from "../components/Header/Header";
 import "../style/index.css";
 import "../style/fonts.css";
+import { Button } from "../components/Button";
+import { Input } from "../components/Input";
 import { Footer } from "../components/footer";
 
+storiesOf("Buttons", module).add("Button primary", () => (
+  <Button isInverted={true} text="Button Primary" />
```
Comment (pggl) [inline src/stories/index.tsx]: there is shorhand for "isInverted={true}" try "isInverted"

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import React, { FC } from "react";
+import classnames from "classnames";
+
+interface IProps {
+  isInverted?: boolean;
+  text: string;
```
Comment (pggl) [inline src/components/Button.tsx]: use children for this

**Review by pggl — CHANGES_REQUESTED** [review]

File: tailwind.js [hunk 16]
```diff
@@ -54,7 +54,8 @@ let colors = {
...
+  "dark-black": "1f1f1f"
```
Comment (pggl) [inline tailwind.js]: missing "#"

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+interface IProps {
+  isInverted?: boolean;
+}
+
+export const Button: FC<IProps> = ({ isInverted = false, children }) => (
+  <button
+    className={classnames(
+      "mt-1 h-10 w-full font-lato text-base hover:shadow-lg hover:bg-dark-black active:opacity-50 mr-20 font-bold  border-solid border border-black rounded-xl",
+      {
+        [`${isInverted ? "bg-black text-white" : "bg-white text-black"}`]: true
```
Comment (pggl) [inline src/components/Button.tsx]: wrong hover for inverted

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Input.tsx [hunk 18]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from "react";
+
+interface IProps {
+  name: string;
+  placeholder: string;
+}
+
+export const Input: FC<IProps> = ({ name, placeholder }) => (
+  <div>
+    <label className="text-sm text-grey font-lato mb-2 mt-3">{name}</label>
```
Comment (pggl) [inline src/components/Input.tsx]: wrong size and mb

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+import classnames from "classnames";
+
+interface IProps {
+  isInverted?: boolean;
+}
+
+export const Button: FC<IProps> = ({ isInverted = false, children }) => (
+  <button
+    className={classnames(
+      "mt-1 h-10 w-full font-lato text-base hover:shadow-lg active:opacity-50 mr-20 font-bold  border-solid border border-black rounded-xl",
```
Comment (garySiladi) [inline src/components/Button.tsx]: after the `font-bold` word you have double-space instead of one space

**Review by garySiladi — CHANGES_REQUESTED** [review]
1 cosmetic remark

**Comment by garySiladi** [comment]
just that one thing ^

**Review by garySiladi — APPROVED** [review]
from my side ok, check with @pggl as well

**Review by pggl — APPROVED** [review]

**Pull request merged**